src/hp/netcheck.rs

```diff
@@ -793,21 +798,20 @@ impl ReportState {
             }
             return None;
         }
-        if !self.sent_hair_check {
-            return None;
-        }
-
-        let hair_timeout = self.hair_timeout.take().expect("should have been started");
-
-        tokio::select! {
-            biased;
-            _ = self.got_hair_stun.recv() => {
-                Some(true)
-            }
-            _ = hair_timeout => {
-                debug!("hair_check timeout");
-                Some(false)
+        match self.hair_timeout {
```

Comment:
doesn't clippy want this as an if-let?  I don't mind though

Reply:
it seems happy 😅